src/quantum/azext_quantum/operations/job.py

```diff
@@ -0,0 +1,240 @@
+
+    args.append(f"-property:ExecutionTarget={target.target_id}")
+
+    if project:
+        args.append("--project")
+        args.append(project)
+
+    logger.debug("Building project with arguments:")
+    logger.debug(args)
+
+    import subprocess
+    result = subprocess.run(args, stdout=subprocess.PIPE, check=False)
+
+    if result.returncode == 0:
+        return {'result': 'ok'}
+
+    raise CLIError("Failed to compile program.")
+
+
+def _generate_submit_args(program_args, ws, target, token, project, job_name, shots, storage):
+    """ Generates the list of arguments for calling submit on a Q# project """
+
+    args = ["dotnet", "run", "--no-build"]
+
+    if project:
+        args.append("--project")
+        args.append(project)
+
+    args.append("--")
+    args.append("submit")
+
+    args.append("--subscription")
+    args.append(ws.subscription)
+
+    args.append("--resource-group")
+    args.append(ws.resource_group)
+
+    args.append("--workspace")
+    args.append(ws.name)
+
+    args.append("--target")
+    args.append(target.target_id)
+
+    args.append("--output")
+    args.append("Id")
+
+    if job_name:
+        args.append("--job-name")
+        args.append(job_name)
+
+    if shots:
+        args.append("--shots")
+        args.append(shots)
+
+    if storage:
+        args.append("--storage")
+        args.append(storage)
+
+    args.append("--aad-token")
+    args.append(token)
+
+    args.append("--base-uri")
+    args.append(base_url(ws.location))
+
+    args.extend(program_args)
+
+    logger.debug("Running project with arguments:")
+    logger.debug(args)
+
+    return args
+
+
+def submit(cmd, program_args, resource_group_name=None, workspace_name=None, location=None,
+           target_id=None, project=None, job_name=None, shots=None, storage=None, no_build=False):
+    """
+    Submit a Q# project to run on Azure Quantum.
+    """
+
+    # We first build and then call run.
+    # Can't call run directly because it fails to understand the
+    # `ExecutionTarget` property when passed in the command line
+    if not no_build:
+        build(cmd, target_id=target_id, project=project)
+
+    logger.info("Project built successfully.")
+
+    ws = WorkspaceInfo(cmd, resource_group_name, workspace_name, location)
+    target = TargetInfo(cmd, target_id)
+    token = _get_data_credentials(cmd.cli_ctx, ws.subscription).get_token().token
+
+    args = _generate_submit_args(program_args, ws, target, token, project, job_name, shots, storage)
+
+    import subprocess
+    result = subprocess.run(args, stdout=subprocess.PIPE, check=False)
+
+    if result.returncode == 0:
+        job_id = result.stdout.decode('ascii').strip()
+        return show(cmd, job_id, resource_group_name, workspace_name)
+
+    raise CLIError("Failed to submit job.")
+
+
+def _parse_blob_url(url):
+    from urllib.parse import urlparse
+    o = urlparse(url)
+
+    account_name = o.netloc.split('.')[0]
+    container = o.path.split('/')[-2]
```

Comment:
Dangerous code. It may raise IndexError: list index out of range.

Reply:
That's a very good point. While I won't change the implementation, I'll add protection in case this happens.